File: lib/server/InputServiceBody.js


//"6,B+E.9,A..-/,O,D,(,./(-A.>*B*K"
class InputServiceBody {

  constructor(clientSeq, WorkerName, ServiceName, InVal, Operation, AppLoginID, AppLoginPswd, OTP, AprStat, AprSeq, MakerDt, TimeOut) { //If model is class, you must define init() func. Struct's don't need
    this.CltVersion = glb_sv.configInfo.build_version ? glb_sv.configInfo.build_version : '3.0.0';
    // Nếu đã login thì mặc định set 2 tham số này từ userInfo
    // if let userInfo = Global.shared.userInfo {
    //     self.AppLoginID = userInfo.c0
    //     self.AppLoginPswd = userInfo.appLoginPassword // Encrypt  đã được gán sau khi login thành công
    //     self.AprID = self.AppLoginID
    //     self.AcntNo = userInfo.c0
    // }

    this.ClientSeq = clientSeq;//Utils.getClientSequence() ; //Client seq
    this.SecCode = glb_sv.activeCode || '---';//Sec code
    this.WorkerName = WorkerName;//Worker name values
    this.ServiceName = ServiceName;//Service name values
    this.TimeOut = TimeOut || 15; //TimeOut values
    this.MWLoginID = Platform.OS === 'android' ? 'Android' : 'IOS';
    this.MWLoginPswd = Platform.OS === 'android' ? ',+A,3-)-C.*,6,9,=+F*K.N*M.=+)+J,004' : "4,D+M,**K/O--+D-D-F-B,E+A,E,B,7";
    this.AppLoginID = AppLoginID || glb_sv.objShareGlb['sessionInfo']['userID'] || '';   //AppLoginID values
    this.AppLoginPswd = AppLoginPswd || glb_sv.objShareGlb['sessionInfo']['passID'] || '';   //AppLoginPswd values (Password has been encrypt)
    this.ClientSentTime = "0";  //ClientSentTime values
    this.Lang = glb_sv.language || "VI";   //Client language
    this.MdmTp = Platform.OS === 'android' ? '03' : '04';  //Media type
    this.InVal = InVal;
    this.TotInVal = InVal.length
    this.AprStat = AprStat || "N";   //Approve status
    this.Operation = Operation;   //Q: Query, I: Insert, U: Update, :anguished: Delete, E: Export, P: Print
    this.CustMgnBrch = "";     //Customer management branch
    this.CustMgnAgc = "";   //Customer management Agency
    this.BrkMgnBrch = "";    //BrokerInfo management branch
    this.BrkMgnAgc = "";    //BrokerInfo management Agency
    this.LoginBrch = '';    //Login branch
    this.LoginAgnc = '';    //Login Agency
  
    this.AprSeq = AprSeq || '';
    this.MakerDt = MakerDt || '';
    this.AprIP = "";
    this.AprID = "";
    this.AprAmt = "";
  
    this.IPPrivate = "192.168.1.113";
  
    // #if os(iOS)
    //   PCName:String = DeviceInfoUtils.instance.name + DeviceInfoUtils.instance.systemName
    // #elseif os(OSX)
    //   PCName:String = Host.current().localizedName!
    // #endif
  
    this.Otp = OTP ? OTP : (glb_sv.objShareGlb['sessionInfo']['Otp'] || "")

    this.AcntNo = '';
    this.SubNo = '';
    this.BankCd = '';
    this.PCName = glb_sv.deviceName || '';
    // this.IPPrivate = '1.1.1.1';
    // this.IPPublic = '1.1.1.1';
    this.SessionID = glb_sv.objShareGlb.sessionInfo.sessionId || '';
  }
}

export default InputServiceBody;

File: lib/input_service_body.dart

import 'dart:convert';

class InputServiceBody {
    String cltVersion;
    int clientSeq;
    String secCode;
    String workerName;
    String serviceName;
    int timeOut;
    String mwLoginId;
    String mwLoginPswd;
    String appLoginId;
    String appLoginPswd;
    String clientSentTime;
    String lang;
    String mdmTp;
    List<String> inVal;
    int totInVal;
    String aprStat;
    String operation;
    String custMgnBrch;
    String custMgnAgc;
    String brkMgnBrch;
    String brkMgnAgc;
    String loginBrch;
    String loginAgnc;
    String aprSeq;
    String makerDt;
    String aprIp;
    String aprId;
    String aprAmt;
    String ipPrivate;
    String otp;
    String acntNo;
    String subNo;
    String bankCd;
    String pcName;
    String sessionId;

    InputServiceBody({
        required this.cltVersion,
        required this.clientSeq,
        required this.secCode,
        required this.workerName,
        required this.serviceName,
        required this.timeOut,
        required this.mwLoginId,
        required this.mwLoginPswd,
        required this.appLoginId,
        required this.appLoginPswd,
        required this.clientSentTime,
        required this.lang,
        required this.mdmTp,
        required this.inVal,
        required this.totInVal,
        required this.aprStat,
        required this.operation,
        required this.custMgnBrch,
        required this.custMgnAgc,
        required this.brkMgnBrch,
        required this.brkMgnAgc,
        required this.loginBrch,
        required this.loginAgnc,
        required this.aprSeq,
        required this.makerDt,
        required this.aprIp,
        required this.aprId,
        required this.aprAmt,
        required this.ipPrivate,
        required this.otp,
        required this.acntNo,
        required this.subNo,
        required this.bankCd,
        required this.pcName,
        required this.sessionId,
    });

    factory InputServiceBody.fromRawJson(String str) => InputServiceBody.fromJson(json.decode(str));

    String toRawJson() => json.encode(toJson());

    factory InputServiceBody.fromJson(Map<String, dynamic> json) => InputServiceBody(
        cltVersion: json["CltVersion"],
        clientSeq: json["ClientSeq"],
        secCode: json["SecCode"],
        workerName: json["WorkerName"],
        serviceName: json["ServiceName"],
        timeOut: json["TimeOut"],
        mwLoginId: json["MWLoginID"],
        mwLoginPswd: json["MWLoginPswd"],
        appLoginId: json["AppLoginID"],
        appLoginPswd: json["AppLoginPswd"],
        clientSentTime: json["ClientSentTime"],
        lang: json["Lang"],
        mdmTp: json["MdmTp"],
        inVal: List<String>.from(json["InVal"].map((x) => x)),
        totInVal: json["TotInVal"],
        aprStat: json["AprStat"],
        operation: json["Operation"],
        custMgnBrch: json["CustMgnBrch"],
        custMgnAgc: json["CustMgnAgc"],
        brkMgnBrch: json["BrkMgnBrch"],
        brkMgnAgc: json["BrkMgnAgc"],
        loginBrch: json["LoginBrch"],
        loginAgnc: json["LoginAgnc"],
        aprSeq: json["AprSeq"],
        makerDt: json["MakerDt"],
        aprIp: json["AprIP"],
        aprId: json["AprID"],
        aprAmt: json["AprAmt"],
        ipPrivate: json["IPPrivate"],
        otp: json["Otp"],
        acntNo: json["AcntNo"],
        subNo: json["SubNo"],
        bankCd: json["BankCd"],
        pcName: json["PCName"],
        sessionId: json["SessionID"],
    );

    Map<String, dynamic> toJson() => {
        "CltVersion": cltVersion,
        "ClientSeq": clientSeq,
        "SecCode": secCode,
        "WorkerName": workerName,
        "ServiceName": serviceName,
        "TimeOut": timeOut,
        "MWLoginID": mwLoginId,
        "MWLoginPswd": mwLoginPswd,
        "AppLoginID": appLoginId,
        "AppLoginPswd": appLoginPswd,
        "ClientSentTime": clientSentTime,
        "Lang": lang,
        "MdmTp": mdmTp,
        "InVal": List<dynamic>.from(inVal.map((x) => x)),
        "TotInVal": totInVal,
        "AprStat": aprStat,
        "Operation": operation,
        "CustMgnBrch": custMgnBrch,
        "CustMgnAgc": custMgnAgc,
        "BrkMgnBrch": brkMgnBrch,
        "BrkMgnAgc": brkMgnAgc,
        "LoginBrch": loginBrch,
        "LoginAgnc": loginAgnc,
        "AprSeq": aprSeq,
        "MakerDt": makerDt,
        "AprIP": aprIp,
        "AprID": aprId,
        "AprAmt": aprAmt,
        "IPPrivate": ipPrivate,
        "Otp": otp,
        "AcntNo": acntNo,
        "SubNo": subNo,
        "BankCd": bankCd,
        "PCName": pcName,
        "SessionID": sessionId,
    };
}
